src/StreamrClient.js

```diff
@@ -435,10 +437,15 @@ export default class StreamrClient extends EventEmitter {
         })
     }
 
-    _requestPublish(streamId, data, apiKey, sessionToken) {
-        const request = new PublishRequest(streamId, apiKey, sessionToken, data)
+    _requestPublish(streamId, data, timestamp, apiKey, sessionToken) {
+        const request = new PublishRequest(streamId, apiKey, sessionToken, data, timestamp)
         debug('_requestResend: %o', request)
-        this.connection.send(request)
+        if (this.signer) {
+            return this.signer.getSignedPublishRequest(request).then((signedRequest) => {
```

Comment:
To avoid repeating `this.connection.send` (and fix the `debug`) you could:

```
let request = new PublishRequest(streamId, apiKey, sessionToken, data, timestamp)
if (this.signer) {
    request = await this.signer.getSignedPublishRequest(request)
}
debug('_requestPublish: %o', request)
return this.connection.send(request)